src/main/java/io/github/jhipster/online/mdc/logging/MDCInterceptor.java

```diff
@@ -0,0 +1,23 @@
+package io.github.jhipster.online.mdc.logging;
+
+import io.github.jhipster.online.security.SecurityUtils;
+import org.slf4j.MDC;
+import org.springframework.web.servlet.AsyncHandlerInterceptor;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class MDCInterceptor implements AsyncHandlerInterceptor {
```

Comment:
I did not know about `AsyncHandlerInterceptor `so I'm not sure whether how this behaves in case of an exception thrown by controller or underlying services. In the past, I used a servlet filter and I did call `MDC.remove` in catch and finally blocks.

The `afterConcurrentHandlingStarted`'s javadoc says:

> A typical use of this method would be to clean up thread-local variables

It seems to me that it should cleanup MDC, have you tested an exception to see how it logs?

Also in my implementation, I remember having defined an `AsyncTaskExecutor` to propagate the MDC because I had issues with `MODE_INHERITABLETHREADLOCAL` but it was 4 years ago....

Reply:
@gmarziou : Okay, I've added the MDC cleanup to the `afterConcurrentHandlingStarted` method instead. You are right that is more appropriate I think and also I've tested with an exception; here's the typical stack trace of an exception that occurs within a Async method (the `createGitProviderRepository` method in `GitHubService.java` class). 

```
2020-07-29 12:45:22.757  WARN 25622 --- [ XNIO-1 task-13] o.z.problem.spring.common.AdviceTraits   : Forbidden: Access is denied
2020-07-29 12:45:22 [XNIO-1 task-13] WARN  o.z.p.spring.common.AdviceTraits - User:anonymousUser - Forbidden: Access is denied
2020-07-29 12:45:42 [XNIO-1 task-27] ERROR i.g.j.online.web.rest.GitResource - User:sudharaka - OAuth2 token could not saved: 
org.springframework.web.client.HttpClientErrorException$NotFound: 404 Not Found: [{"error":"Not Found"}]
        at org.springframework.web.client.HttpClientErrorException.create(HttpClientErrorException.java:113)
        at org.springframework.web.client.DefaultResponseErrorHandler.handleError(DefaultResponseErrorHandler.java:170)
        at org.springframework.web.client.DefaultResponseErrorHandler.handleError(DefaultResponseErrorHandler.java:112)
        at org.springframework.web.client.ResponseErrorHandler.handleError(ResponseErrorHandler.java:63)
        at org.springframework.web.client.RestTemplate.handleResponse(RestTemplate.java:782)
        at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:740)
        at org.springframework.web.client.RestTemplate.execute(RestTemplate.java:674)
        at org.springframework.web.client.RestTemplate.postForEntity(RestTemplate.java:449)
        at io.github.jhipster.online.web.rest.GitResource.saveToken(GitResource.java:126)
        at io.github.jhipster.online.web.rest.GitResource$$FastClassBySpringCGLIB$$ca87e103.invoke(<generated>)
        at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
```

Seems it looks good right? :thinking: